server/src/com/thoughtworks/go/server/messaging/plugin/StageStatusPluginNotifier.java

```diff
@@ -16,49 +16,258 @@
+import com.thoughtworks.go.plugin.access.notification.NotificationPluginRegistry;
 import com.thoughtworks.go.server.domain.StageStatusListener;
+import com.thoughtworks.go.server.service.PipelineService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import java.sql.Timestamp;
 import java.text.SimpleDateFormat;
-import java.util.LinkedHashMap;
-import java.util.Map;
+import java.util.*;
 
 @Component
 public class StageStatusPluginNotifier implements StageStatusListener {
     public static final String DATE_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
 
+    private NotificationPluginRegistry notificationPluginRegistry;
+    private PipelineService pipelineService;
     private PluginNotificationQueue pluginNotificationQueue;
 
     @Autowired
-    public StageStatusPluginNotifier(PluginNotificationQueue pluginNotificationQueue) {
+    public StageStatusPluginNotifier(NotificationPluginRegistry notificationPluginRegistry, PipelineService pipelineService, PluginNotificationQueue pluginNotificationQueue) {
+        this.notificationPluginRegistry = notificationPluginRegistry;
+        this.pipelineService = pipelineService;
         this.pluginNotificationQueue = pluginNotificationQueue;
     }
 
     @Override
     public void stageStatusChanged(final Stage stage) {
-        Map data = createRequestDataMap(stage);
+        if (notificationPluginRegistry.isAnyPluginInterestedIn(NotificationExtension.STAGE_STATUS_CHANGE_NOTIFICATION)) {
+            Map data = createRequestDataMap(stage);
 
-        pluginNotificationQueue.post(new PluginNotificationMessage(NotificationExtension.STAGE_STATUS_CHANGE_NOTIFICATION, data));
+            pluginNotificationQueue.post(new PluginNotificationMessage(NotificationExtension.STAGE_STATUS_CHANGE_NOTIFICATION, data));
+        }
     }
 
     Map createRequestDataMap(Stage stage) {
-        Map data = new LinkedHashMap();
+        Map<String, Object> data = new LinkedHashMap<String, Object>();
         data.put("pipeline-name", stage.getIdentifier().getPipelineName());
-        data.put("pipeline-counter", stage.getIdentifier().getPipelineCounter());
+        data.put("pipeline-counter", new Integer(stage.getIdentifier().getPipelineCounter()).toString());
         data.put("stage-name", stage.getIdentifier().getStageName());
         data.put("stage-counter", stage.getIdentifier().getStageCounter());
-        data.put("stage-state", stage.getState());
-        data.put("stage-result", stage.getResult());
-        data.put("create-time", timestampToString(stage.getCreatedTime()));
-        data.put("last-transition-time", timestampToString(stage.getLastTransitionedTime()));
+        data.put("stage-state", stage.getState().toString());
+        data.put("stage-result", stage.getResult().toString());
+
+        Pipeline pipeline = pipelineService.fullPipelineById(stage.getPipelineId());
```

Comment:
This is not a cached call. I think it can become problematic. Can you use [pipelineWithModsByStageId](https://github.com/gocd/gocd/blob/master/server/src/com/thoughtworks/go/server/service/PipelineService.java#L77) instead?